src/lib/es5.d.ts

```diff
@@ -12,7 +12,7 @@ declare var Infinity: number;
 declare function eval(x: string): any;
 
 /**
-  * Converts A string to an integer.
+  * Converts a string to an integer.
   * @param s A string to convert into a number.
   * @param radix A value between 2 and 36 that specifies the base of the number in numString.
   * If this argument is not supplied, strings with a prefix of '0x' are considered hexadecimal.

```

Comment:
Wait, it says below `All other strings are considered decimal.` -- what???

```javascript
parseInt(010)  === 8
```

The warning about `0`-leading strings should be big and in front. The `0x` quirk is rather minor in comparison, and the `a`/`A` quirk is totally insifnigicant.

BTW, there's a specific <a href="https://eslint.org/docs/rules/radix">ESLint rule</a> to warn against `parseInt` with octal format.

Reply:
In all modern browsers and node `parseInt('010')  === 10`.
This change was introduced in ES5 see [details on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#ECMAScript_5_removes_octal_interpretation)

Reply:
Two engines still valid in JS world, WSH and HTA, have that behaviour.

Although this file `es5.d.ts` may be taken to imply ES5-only types, it is also used for ES3.